File: src/Publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Float64.h"
#include <sstream>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "publisher");
  ros::NodeHandle n;
  ros::Publisher pub = n.advertise<std_msgs::Float64>("counter", 1);
  ros::Rate loop_rate(2);

  int count = 0;
  while (ros::ok())
  {
    std_msgs::Float64 number;
    number.data = (float)count;
    ROS_INFO("%f", number.data);

    pub.publish(number);
    ros::spinOnce();
    loop_rate.sleep();
    ++count;
  }
  return 0;
}

File: src/Publisher.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64

if __name__ == "__main__":

  rospy.init_node('publisher')
  pub = rospy.Publisher('counter', Float64, queue_size=1)
  rate = rospy.Rate(2) # 2 Hz
  
  count = 0
  
  while not rospy.is_shutdown():
    
    result = Float64()
    result.data = count
    rospy.loginfo(str(result.data))
    pub.publish(result)
    count = count + 1
    rate.sleep()
